tests/testthat/test-label-number.r

```diff
@@ -118,3 +125,43 @@ test_that("precision ignores Inf and NA", {
   expect_equal(precision(c(NA, Inf, -Inf)), 1)
   expect_equal(precision(c(1, NA)), 1)
 })
+
+
+# scale_cut ---------------------------------------------------------------
+
+test_that("accuracy computed on individual scales", {
+  x <- c(0, 250, 500, 750, 1250, 1500) * 1e3
+  expect_equal(
+    number(x, scale_cut = cut_short_scale()),
+    c("0", "250K", "500K", "750K", "1.25M", "1.50M")
+  )
+})
+
+test_that("zero and special values don't get units", {
+  x <- c(0, NA, Inf, -Inf)
+  expect_equal(
+    number(x, scale_cut = cut_short_scale()),
+    c("0", NA, "Inf", "-Inf")
+  )
+})
+
+test_that("scale applied before scale_cut", {
+  expect_equal(number(500, scale = 2, scale_cut = cut_short_scale()), "1K")
+})
+
+test_that("cut_si() adds space before unit", {
+  expect_equal(number(c(1, 1000), scale_cut = cut_si("m")), c("1 m", "1 km"))
+})
+
```

Comment:
I think we should test how `cut_si("m")` handles zero. It seems like an edge case due to how `cut_si()` bakes the units into the scale suffix. IMHO the expected result should be "0 m".

Reply:
Ok, should be fixed now.